Automation/wdio/practice/src/ui/services/products/addNewProduct.service.ts

```diff
@@ -0,0 +1,32 @@
+import { generateNewProduct } from '../../../data/products/generateProduct.js';
+import { IProduct } from '../../../data/types/product.types.js';
+import { logStep } from '../../../utils/report/decorator.js';
+import { AddNewProductPage } from '../../pages/products/addNewProduct.page.js';
+import { ProductsPage } from '../../pages/products/products.page.js';
+
+export class AddProductService {
+  constructor(
+    private productsPage = new ProductsPage(),
+    private addNewProductPage = new AddNewProductPage()
+  ) {}
+
+  @logStep('Fill product inputs')
+  async fillProductInputs(product: Partial<IProduct>) {
+    await this.addNewProductPage.fillInputs(product);
+  }
+
+  @logStep('Save new product')
+  async save() {
+    await this.addNewProductPage.clickOnSaveButton();
+  }
+
+  @logStep('Create product')
+  async create(product?: IProduct) {
+    await this.fillProductInputs(product ?? generateNewProduct());
+    await this.save();
+    await this.addNewProductPage.waitForSpinnerToHide();
+    await this.productsPage.waitForOpened();
+    await this.productsPage.closePopup();
```

Comment:
I belive that method should be called verifyAndCloseNotification(text: string). Service should handle notifications and it's expected messages, not page object